unexpectedly reached end of fileSIGILL: illegal instruction

I'm trying to use many models, but none of them are working when I use the /chat/completions page. Here's what I'm getting with gpt4all:

I'm using Postman with this query:

`{
    "model": "gpt",
    "messages": [
        {
            "role": "user",
            "content": "How are you?"
        }
    ],
    "temperature": 0.9
}`

> Starting LocalAI using 4 threads, with models path: /models
 ┌───────────────────────────────────────────────────┐ 
 │                   Fiber v2.45.0                   │ 
 │               http://127.0.0.1:8080               │ 
 │       (bound on host 0.0.0.0 and port 8080)       │ 
 │                                                   │ 
 │ Handlers ............ 17  Processes ........... 1 │ 
 │ Prefork ....... Disabled  PID ............... 164 │ 
 └───────────────────────────────────────────────────┘ 
llama.cpp: loading model from /models/gpt
failed unexpectedly reached end of fileSIGILL: illegal instruction
PC=0xa893c4 m=5 sigcode=2
signal arrived during cgo execution
instruction bytes: 0xc4 0xc3 0x7d 0x39 0x86 0xd8 0x13 0x0 0x0 0x1 0x49 0x89 0x86 0xb8 0x14 0x0
goroutine 18 [syscall]:
runtime.cgocall(0x97b760, 0xc0001ae920)
        /usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc0001ae8f8 sp=0xc0001ae8c0 pc=0x44a4fc
github.com/nomic-ai/gpt4all/gpt4all-bindings/golang._Cfunc_load_gptjllama_model(0x7fa4e8002550, 0x4)
        _cgo_gotypes.go:137 +0x4d fp=0xc0001ae920 sp=0xc0001ae8f8 pc=0x584f0d
github.com/nomic-ai/gpt4all/gpt4all-bindings/golang.New({0xc000236650, 0xb}, {0xc000209080, 0x2, 0x1?})
        /build/gpt4all/gpt4all-bindings/golang/gpt4all.go:35 +0x145 fp=0xc0001ae978 sp=0xc0001ae920 pc=0x585225
github.com/go-skynet/LocalAI/pkg/model.gpt4allLM.func1({0xc000236650?, 0xbca30f?})
        /build/pkg/model/initializers.go:110 +0x2a fp=0xc0001ae9b0 sp=0xc0001ae978 pc=0x607d4a
github.com/go-skynet/LocalAI/pkg/model.(*ModelLoader).LoadModel(0xc0001a66c0, {0xc000236568, 0x3}, 0xc000024ca0)
        /build/pkg/model/loader.go:127 +0x1fe fp=0xc0001aeaa8 sp=0xc0001ae9b0 pc=0x609c9e
github.com/go-skynet/LocalAI/pkg/model.(*ModelLoader).BackendLoader(0xc0001a66c0, {0xbba207, 0xd}, {0xc000236568, 0x3}, {0xc000014590, 0x1, 0x1}, 0x4)
        /build/pkg/model/initializers.go:150 +0x7d2 fp=0xc0001aeb70 sp=0xc0001aeaa8 pc=0x608752
github.com/go-skynet/LocalAI/pkg/model.(*ModelLoader).GreedyLoader(0xc0001a66c0, {0xc000236568, 0x3}, {0xc000014590, 0x1, 0x1}, 0x0?)
        /build/pkg/model/initializers.go:183 +0x2e5 fp=0xc0001aeca8 sp=0xc0001aeb70 pc=0x608ca5
github.com/go-skynet/LocalAI/api.ModelInference({_, _}, _, {{{0xc000236568, 0x3}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...}, ...}, ...)
        /build/api/prediction.go:218 +0x145 fp=0xc0001aef58 sp=0xc0001aeca8 pc=0x8d8325
github.com/go-skynet/LocalAI/api.ComputeChoices({0xc0002400c0, 0xba}, 0xc00014ab40, 0xc000115b80, 0xc000208f50?, 0xbf46b8, 0x4?)
        /build/api/prediction.go:517 +0x138 fp=0xc0001af708 sp=0xc0001aef58 pc=0x8db9f8
github.com/go-skynet/LocalAI/api.chatEndpoint.func2(0xc000132840)
        /build/api/openai.go:361 +0x8ec fp=0xc0001af8c8 sp=0xc0001af708 pc=0x8d35cc
github.com/gofiber/fiber/v2.(*App).next(0xc000138d80, 0xc000132840)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/router.go:144 +0x1bf fp=0xc0001af970 sp=0xc0001af8c8 pc=0x88993f
github.com/gofiber/fiber/v2.(*Ctx).Next(0xc0000a0330?)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/ctx.go:913 +0x53 fp=0xc0001af990 sp=0xc0001af970 pc=0x874f13
github.com/gofiber/fiber/v2/middleware/cors.New.func1(0xc000132840)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/middleware/cors/cors.go:162 +0x3da fp=0xc0001afa98 sp=0xc0001af990 pc=0x88f75a
github.com/gofiber/fiber/v2.(*Ctx).Next(0x14?)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/ctx.go:910 +0x43 fp=0xc0001afab8 sp=0xc0001afa98 pc=0x874f03
github.com/gofiber/fiber/v2/middleware/recover.New.func1(0xb3c360?)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/middleware/recover/recover.go:43 +0xcb fp=0xc0001afb30 sp=0xc0001afab8 pc=0x89638b
github.com/gofiber/fiber/v2.(*App).next(0xc000138d80, 0xc000132840)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/router.go:144 +0x1bf fp=0xc0001afbd8 sp=0xc0001afb30 pc=0x88993f
github.com/gofiber/fiber/v2.(*App).handler(0xc000138d80, 0x50ce57?)
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/router.go:171 +0x87 fp=0xc0001afc38 sp=0xc0001afbd8 pc=0x889b87
github.com/gofiber/fiber/v2.(*App).handler-fm(0xc0000a0000?)
        <autogenerated>:1 +0x2c fp=0xc0001afc58 sp=0xc0001afc38 pc=0x88edac
github.com/valyala/fasthttp.(*Server).serveConn(0xc0001fc400, {0xc79c80?, 0xc00008e008})
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/server.go:2365 +0x11d3 fp=0xc0001afec8 sp=0xc0001afc58 pc=0x80fb33
github.com/valyala/fasthttp.(*Server).serveConn-fm({0xc79c80?, 0xc00008e008?})
        <autogenerated>:1 +0x39 fp=0xc0001afef0 sp=0xc0001afec8 pc=0x81f3f9
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc000117680, 0xc00009e000)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/workerpool.go:224 +0xa9 fp=0xc0001affa0 sp=0xc0001afef0 pc=0x81b629
github.com/valyala/fasthttp.(*workerPool).getCh.func1()
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/workerpool.go:196 +0x38 fp=0xc0001affe0 sp=0xc0001affa0 pc=0x81b398
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0001affe8 sp=0xc0001affe0 pc=0x4ac9c1
created by github.com/valyala/fasthttp.(*workerPool).getCh
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/workerpool.go:195 +0x1b0
goroutine 1 [IO wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc0001cd418 sp=0xc0001cd3f8 pc=0x47e016
runtime.netpollblock(0x7fa500316548?, 0x449b8f?, 0x0?)
        /usr/local/go/src/runtime/netpoll.go:527 +0xf7 fp=0xc0001cd450 sp=0xc0001cd418 pc=0x476977
internal/poll.runtime_pollWait(0x7fa5002ea418, 0x72)
        /usr/local/go/src/runtime/netpoll.go:306 +0x89 fp=0xc0001cd470 sp=0xc0001cd450 pc=0x4a76c9
internal/poll.(*pollDesc).wait(0xc000166b00?, 0x4?, 0x0)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x32 fp=0xc0001cd498 sp=0xc0001cd470 pc=0x51ae92
internal/poll.(*pollDesc).waitRead(...)
        /usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc000166b00)
        /usr/local/go/src/internal/poll/fd_unix.go:614 +0x2bd fp=0xc0001cd540 sp=0xc0001cd498 pc=0x52079d
net.(*netFD).accept(0xc000166b00)
        /usr/local/go/src/net/fd_unix.go:172 +0x35 fp=0xc0001cd5f8 sp=0xc0001cd540 pc=0x5a8c15
net.(*TCPListener).accept(0xc000012828)
        /usr/local/go/src/net/tcpsock_posix.go:148 +0x25 fp=0xc0001cd620 sp=0xc0001cd5f8 pc=0x5beea5
net.(*TCPListener).Accept(0xc000012828)
        /usr/local/go/src/net/tcpsock.go:297 +0x3d fp=0xc0001cd650 sp=0xc0001cd620 pc=0x5bdf9d
github.com/valyala/fasthttp.acceptConn(0xc0001fc400, {0xc773e0, 0xc000012828}, 0xc0001cd848)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/server.go:1930 +0x62 fp=0xc0001cd730 sp=0xc0001cd650 pc=0x80e002
github.com/valyala/fasthttp.(*Server).Serve(0xc0001fc400, {0xc773e0?, 0xc000012828})
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/server.go:1823 +0x4f4 fp=0xc0001cd878 sp=0xc0001cd730 pc=0x80d614
github.com/gofiber/fiber/v2.(*App).Listen(0xc000138d80, {0xbb0b88?, 0x7?})
        /go/pkg/mod/github.com/gofiber/fiber/v2@v2.45.0/listen.go:82 +0x110 fp=0xc0001cd8d8 sp=0xc0001cd878 pc=0x880a30
main.main.func1(0xc0001cdbc8?)
        /build/main.go:96 +0x2f9 fp=0xc0001cd9b8 sp=0xc0001cd8d8 pc=0x909919
github.com/urfave/cli/v2.(*Command).Run(0xc0001ba160, 0xc00006a800, {0xc000024060, 0x2, 0x2})
        /go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/command.go:274 +0x9eb fp=0xc0001cdc58 sp=0xc0001cd9b8 pc=0x8f786b
github.com/urfave/cli/v2.(*App).RunContext(0xc0001b6000, {0xc776b8?, 0xc000028078}, {0xc000024060, 0x2, 0x2})
        /go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/app.go:332 +0x616 fp=0xc0001cdcc8 sp=0xc0001cdc58 pc=0x8f4676
github.com/urfave/cli/v2.(*App).Run(...)
        /go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/app.go:309
main.main()
        /build/main.go:100 +0xbae fp=0xc0001cdf80 sp=0xc0001cdcc8 pc=0x90954e
runtime.main()
        /usr/local/go/src/runtime/proc.go:250 +0x207 fp=0xc0001cdfe0 sp=0xc0001cdf80 pc=0x47dbe7
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0001cdfe8 sp=0xc0001cdfe0 pc=0x4ac9c1
goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000040fb0 sp=0xc000040f90 pc=0x47e016
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.forcegchelper()
        /usr/local/go/src/runtime/proc.go:305 +0xb0 fp=0xc000040fe0 sp=0xc000040fb0 pc=0x47de50
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc000040fe8 sp=0xc000040fe0 pc=0x4ac9c1
created by runtime.init.6
        /usr/local/go/src/runtime/proc.go:293 +0x25
goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000041780 sp=0xc000041760 pc=0x47e016
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.bgsweep(0x0?)
        /usr/local/go/src/runtime/mgcsweep.go:278 +0x8e fp=0xc0000417c8 sp=0xc000041780 pc=0x46a22e
runtime.gcenable.func1()
        /usr/local/go/src/runtime/mgc.go:178 +0x26 fp=0xc0000417e0 sp=0xc0000417c8 pc=0x45f4e6
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0000417e8 sp=0xc0000417e0 pc=0x4ac9c1
created by runtime.gcenable
        /usr/local/go/src/runtime/mgc.go:178 +0x6b
goroutine 4 [GC scavenge wait]:
runtime.gopark(0xc000068000?, 0xc701e8?, 0x1?, 0x0?, 0x0?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000041f70 sp=0xc000041f50 pc=0x47e016
runtime.goparkunlock(...)
        /usr/local/go/src/runtime/proc.go:387
runtime.(*scavengerState).park(0x1060e60)
        /usr/local/go/src/runtime/mgcscavenge.go:400 +0x53 fp=0xc000041fa0 sp=0xc000041f70 pc=0x468153
runtime.bgscavenge(0x0?)
        /usr/local/go/src/runtime/mgcscavenge.go:628 +0x45 fp=0xc000041fc8 sp=0xc000041fa0 pc=0x468725
runtime.gcenable.func2()
        /usr/local/go/src/runtime/mgc.go:179 +0x26 fp=0xc000041fe0 sp=0xc000041fc8 pc=0x45f486
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc000041fe8 sp=0xc000041fe0 pc=0x4ac9c1
created by runtime.gcenable
        /usr/local/go/src/runtime/mgc.go:179 +0xaa
goroutine 5 [finalizer wait]:
runtime.gopark(0x1a0?, 0x1061b40?, 0x60?, 0x78?, 0xc000040770?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000040628 sp=0xc000040608 pc=0x47e016
runtime.runfinq()
        /usr/local/go/src/runtime/mfinal.go:193 +0x107 fp=0xc0000407e0 sp=0xc000040628 pc=0x45e527
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0000407e8 sp=0xc0000407e0 pc=0x4ac9c1
created by runtime.createfing
        /usr/local/go/src/runtime/mfinal.go:163 +0x45
goroutine 6 [sleep]:
runtime.gopark(0xf34e8098faa?, 0xc000042788?, 0x5?, 0xd8?, 0xc0001176b0?)
        /usr/local/go/src/runtime/proc.go:381 +0xd6 fp=0xc000042758 sp=0xc000042738 pc=0x47e016
time.Sleep(0x2540be400)
        /usr/local/go/src/runtime/time.go:195 +0x135 fp=0xc000042798 sp=0xc000042758 pc=0x4a9835
github.com/valyala/fasthttp.(*workerPool).Start.func2()
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/workerpool.go:67 +0x56 fp=0xc0000427e0 sp=0xc000042798 pc=0x81aaf6
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc0000427e8 sp=0xc0000427e0 pc=0x4ac9c1
created by github.com/valyala/fasthttp.(*workerPool).Start
        /go/pkg/mod/github.com/valyala/fasthttp@v1.47.0/workerpool.go:59 +0xdd
rax    0x7fa4e8005398
rbx    0xffffffff
rcx    0x20
rdx    0x4c46d8c
rdi    0x7fa4e8005408
rsi    0x16e0
rbp    0x7fa4fbffebf0
rsp    0x7fa4fbffd730
r8     0x7fa4e8003eb0
r9     0x7fa4e8000080
r10    0x4
r11    0x0
r12    0x7fa4fbffec70
r13    0x7fa4fbffedb0
r14    0x7fa4e8003eb0
r15    0xc000080000
rip    0xa893c4
rflags 0x10246
cs     0x33
fs     0x0
gs     0x0

Hey, I have something to add to this. I did run into this issue with an AVX only CPU. I wasn't able to get docker to work but I could build the LocalAI binary using ```CMAKE_ARGS="-DLLAMA_AVX2=OFF" ``` and it didn't have this issue outside of docker. When I try to pass this flag to the docker build the build fails with this instead

```#0 524.3 g++: warning: ggml.o: linker input file unused because linking not done
#0 524.3 g++: warning: utils.o: linker input file unused because linking not done
#0 524.3 /go/pkg/mod/github.com/mattn/go-isatty@v0.0.18/isatty_tcgets.go:7:8: missing go.sum entry for module providing package golang.org/x/sys/unix (imported by github.com/mattn/go-isatty); to add:
#0 524.3 	go get github.com/mattn/go-isatty@v0.0.18
```

Is this an issue with the go docker container?

--- a/Makefile
+++ b/Makefile
@@ -152,7 +152,7 @@ ifeq ($(findstring tts,$(GO_TAGS)),tts)
 	OPTIONAL_GRPC+=backend-assets/grpc/piper
 endif
 
-ALL_GRPC_BACKENDS=backend-assets/grpc/langchain-huggingface
+ALL_GRPC_BACKENDS=backend-assets/grpc/huggingface
 ALL_GRPC_BACKENDS+=backend-assets/grpc/bert-embeddings
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp-noavx
@@ -619,8 +619,8 @@ backend-assets/grpc/gpt4all: sources/gpt4all sources/gpt4all/gpt4all-bindings/go
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" C_INCLUDE_PATH=$(CURDIR)/sources/gpt4all/gpt4all-bindings/golang/ LIBRARY_PATH=$(CURDIR)/sources/gpt4all/gpt4all-bindings/golang/ \
 	$(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o backend-assets/grpc/gpt4all ./backend/go/llm/gpt4all/
 
-backend-assets/grpc/langchain-huggingface: backend-assets/grpc
-	$(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o backend-assets/grpc/langchain-huggingface ./backend/go/llm/langchain/
+backend-assets/grpc/huggingface: backend-assets/grpc
+	$(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o backend-assets/grpc/huggingface ./backend/go/llm/langchain/
 
 backend/cpp/llama/llama.cpp:
 	LLAMA_VERSION=$(CPPLLAMA_VERSION) $(MAKE) -C backend/cpp/llama llama.cpp

--- a/Makefile
+++ b/Makefile
@@ -660,7 +660,8 @@ backend-assets/grpc/llama-cpp-noavx: backend-assets/grpc
 	$(info ${GREEN}I llama-cpp build info:noavx${RESET})
 	cp -rf backend/cpp/llama backend/cpp/llama-noavx
 	$(MAKE) -C backend/cpp/llama-noavx purge
-	CMAKE_ARGS="-DLLAMA_AVX2=OFF" $(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
+	CMAKE_ARGS+=-DLLAMA_AVX2=OFF
+	$(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
 	cp -rfv backend/cpp/llama-noavx/grpc-server backend-assets/grpc/llama-cpp-noavx
 
 backend-assets/grpc/llama-ggml: sources/go-llama.cpp sources/go-llama.cpp/libbinding.a backend-assets/grpc

--- a/Makefile
+++ b/Makefile
@@ -630,7 +630,7 @@ ADDED_CMAKE_ARGS=-Dabsl_DIR=${INSTALLED_LIB_CMAKE}/absl \
 				 -Dutf8_range_DIR=${INSTALLED_LIB_CMAKE}/utf8_range \
 				 -DgRPC_DIR=${INSTALLED_LIB_CMAKE}/grpc \
 				 -DCMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES=${INSTALLED_PACKAGES}/include
-backend/cpp/llama/grpc-server:
+build-llama-cpp-grpc-server:
 # Conditionally build grpc for the llama backend to use if needed
 ifdef BUILD_GRPC_FOR_BACKEND_LLAMA
 	$(MAKE) -C backend/cpp/grpc build
@@ -639,27 +639,29 @@ ifdef BUILD_GRPC_FOR_BACKEND_LLAMA
 	PATH="${INSTALLED_PACKAGES}/bin:${PATH}" \
 	CMAKE_ARGS="${CMAKE_ARGS} ${ADDED_CMAKE_ARGS}" \
 	LLAMA_VERSION=$(CPPLLAMA_VERSION) \
-	$(MAKE) -C backend/cpp/llama grpc-server
+	$(MAKE) -C backend/cpp/${VARIANT} grpc-server
 else
 	echo "BUILD_GRPC_FOR_BACKEND_LLAMA is not defined."
-	LLAMA_VERSION=$(CPPLLAMA_VERSION) $(MAKE) -C backend/cpp/llama grpc-server
+	LLAMA_VERSION=$(CPPLLAMA_VERSION) $(MAKE) -C backend/cpp/${VARIANT} grpc-server
 endif
 
 backend-assets/grpc/llama-cpp: backend-assets/grpc
 	$(info ${GREEN}I llama-cpp build info:standard${RESET})
-	$(MAKE) -C backend/cpp/llama purge
-	$(MAKE) backend/cpp/llama/grpc-server
-	cp -rfv backend/cpp/llama/grpc-server backend-assets/grpc/llama-cpp
+	cp -rf backend/cpp/llama backend/cpp/llama-default
+	$(MAKE) -C backend/cpp/llama-default purge
+	$(MAKE) VARIANT="llama-default" build-llama-cpp-grpc-server
+	cp -rfv backend/cpp/llama-default/grpc-server backend-assets/grpc/llama-cpp
 # TODO: every binary should have its own folder instead, so can have different metal implementations
 ifeq ($(BUILD_TYPE),metal)
-	cp backend/cpp/llama/llama.cpp/build/bin/default.metallib backend-assets/grpc/
+	cp backend/cpp/llama-default/llama.cpp/build/bin/default.metallib backend-assets/grpc/
 endif
 
 backend-assets/grpc/llama-cpp-noavx: backend-assets/grpc
 	$(info ${GREEN}I llama-cpp build info:noavx${RESET})
-	$(MAKE) -C backend/cpp/llama purge
-	CMAKE_ARGS="-DLLAMA_AVX2=OFF" $(MAKE) backend/cpp/llama/grpc-server
-	cp -rfv backend/cpp/llama/grpc-server backend-assets/grpc/llama-cpp-noavx
+	cp -rf backend/cpp/llama backend/cpp/llama-noavx
+	$(MAKE) -C backend/cpp/llama-noavx purge
+	CMAKE_ARGS="-DLLAMA_AVX2=OFF" $(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
+	cp -rfv backend/cpp/llama-noavx/grpc-server backend-assets/grpc/llama-cpp-noavx
 
 backend-assets/grpc/llama-ggml: sources/go-llama.cpp sources/go-llama.cpp/libbinding.a backend-assets/grpc
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" C_INCLUDE_PATH=$(CURDIR)/sources/go-llama.cpp LIBRARY_PATH=$(CURDIR)/sources/go-llama.cpp \

--- a/Makefile
+++ b/Makefile
@@ -155,6 +155,7 @@ endif
 ALL_GRPC_BACKENDS=backend-assets/grpc/langchain-huggingface
 ALL_GRPC_BACKENDS+=backend-assets/grpc/bert-embeddings
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp
+ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp-noavx
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-ggml
 ALL_GRPC_BACKENDS+=backend-assets/grpc/gpt4all
 ALL_GRPC_BACKENDS+=backend-assets/grpc/rwkv
@@ -311,7 +312,7 @@ build: prepare backend-assets grpcs ## Build the project
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" $(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o $(BINARY_NAME) ./
 
 build-minimal:
-	BUILD_GRPC_FOR_BACKEND_LLAMA=true GRPC_BACKENDS=backend-assets/grpc/llama-cpp GO_TAGS=none $(MAKE) build
+	BUILD_GRPC_FOR_BACKEND_LLAMA=true GRPC_BACKENDS="backend-assets/grpc/llama-cpp backend-assets/grpc/llama-cpp-noavx" GO_TAGS=none $(MAKE) build
 
 build-api:
 	BUILD_GRPC_FOR_BACKEND_LLAMA=true BUILD_API_ONLY=true GO_TAGS=none $(MAKE) build
@@ -644,12 +645,21 @@ else
 	LLAMA_VERSION=$(CPPLLAMA_VERSION) $(MAKE) -C backend/cpp/llama grpc-server
 endif
 
-backend-assets/grpc/llama-cpp: backend-assets/grpc backend/cpp/llama/grpc-server
+backend-assets/grpc/llama-cpp: backend-assets/grpc
+	$(info ${GREEN}I llama-cpp build info:standard${RESET})
+	$(MAKE) -C backend/cpp/llama purge
+	$(MAKE) backend/cpp/llama/grpc-server
 	cp -rfv backend/cpp/llama/grpc-server backend-assets/grpc/llama-cpp
 # TODO: every binary should have its own folder instead, so can have different metal implementations
 ifeq ($(BUILD_TYPE),metal)
 	cp backend/cpp/llama/llama.cpp/build/bin/default.metallib backend-assets/grpc/
 endif
+
+backend-assets/grpc/llama-cpp-noavx: backend-assets/grpc
+	$(info ${GREEN}I llama-cpp build info:noavx${RESET})
+	$(MAKE) -C backend/cpp/llama purge
+	CMAKE_ARGS="-DLLAMA_AVX2=OFF" $(MAKE) backend/cpp/llama/grpc-server
+	cp -rfv backend/cpp/llama/grpc-server backend-assets/grpc/llama-cpp-noavx
 
 backend-assets/grpc/llama-ggml: sources/go-llama.cpp sources/go-llama.cpp/libbinding.a backend-assets/grpc
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" C_INCLUDE_PATH=$(CURDIR)/sources/go-llama.cpp LIBRARY_PATH=$(CURDIR)/sources/go-llama.cpp \

--- a/Makefile
+++ b/Makefile
@@ -664,6 +664,17 @@ backend-assets/grpc/llama-cpp-noavx: backend-assets/grpc
 	$(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
 	cp -rfv backend/cpp/llama-noavx/grpc-server backend-assets/grpc/llama-cpp-noavx
 
+backend-assets/grpc/llama-cpp-fallback: backend-assets/grpc
+	$(info ${GREEN}I llama-cpp build info:fallback${RESET})
+	cp -rf backend/cpp/llama backend/cpp/llama-fallback
+	$(MAKE) -C backend/cpp/llama-fallback purge
+	CMAKE_ARGS+=-DLLAMA_F16C=OFF
+	CMAKE_ARGS+=-DLLAMA_AVX512=OFF
+	CMAKE_ARGS+=-DLLAMA_AVX2=OFF
+	CMAKE_ARGS+=-DLLAMA_FMA=OFF
+	$(MAKE) VARIANT="llama-fallback" build-llama-cpp-grpc-server
+	cp -rfv backend/cpp/llama-fallback/grpc-server backend-assets/grpc/llama-cpp-fallback
+
 backend-assets/grpc/llama-ggml: sources/go-llama.cpp sources/go-llama.cpp/libbinding.a backend-assets/grpc
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" C_INCLUDE_PATH=$(CURDIR)/sources/go-llama.cpp LIBRARY_PATH=$(CURDIR)/sources/go-llama.cpp \
 	$(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o backend-assets/grpc/llama-ggml ./backend/go/llm/llama-ggml/

--- a/Makefile
+++ b/Makefile
@@ -156,6 +156,7 @@ ALL_GRPC_BACKENDS=backend-assets/grpc/langchain-huggingface
 ALL_GRPC_BACKENDS+=backend-assets/grpc/bert-embeddings
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp-noavx
+ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-cpp-fallback
 ALL_GRPC_BACKENDS+=backend-assets/grpc/llama-ggml
 ALL_GRPC_BACKENDS+=backend-assets/grpc/gpt4all
 ALL_GRPC_BACKENDS+=backend-assets/grpc/rwkv
@@ -294,6 +295,7 @@ clean: ## Remove build related file
 	rm -rf backend-assets/*
 	$(MAKE) -C backend/cpp/grpc clean
 	$(MAKE) -C backend/cpp/llama clean
+	rm -rf backend/cpp/llama-* || true
 	$(MAKE) dropreplace
 	$(MAKE) protogen-clean
 	rmdir pkg/grpc/proto || true
@@ -312,7 +314,7 @@ build: prepare backend-assets grpcs ## Build the project
 	CGO_LDFLAGS="$(CGO_LDFLAGS)" $(GOCMD) build -ldflags "$(LD_FLAGS)" -tags "$(GO_TAGS)" -o $(BINARY_NAME) ./
 
 build-minimal:
-	BUILD_GRPC_FOR_BACKEND_LLAMA=true GRPC_BACKENDS="backend-assets/grpc/llama-cpp backend-assets/grpc/llama-cpp-noavx" GO_TAGS=none $(MAKE) build
+	BUILD_GRPC_FOR_BACKEND_LLAMA=true GRPC_BACKENDS="backend-assets/grpc/llama-cpp" GO_TAGS=none $(MAKE) build
 
 build-api:
 	BUILD_GRPC_FOR_BACKEND_LLAMA=true BUILD_API_ONLY=true GO_TAGS=none $(MAKE) build
@@ -657,22 +659,17 @@ ifeq ($(BUILD_TYPE),metal)
 endif
 
 backend-assets/grpc/llama-cpp-noavx: backend-assets/grpc
-	$(info ${GREEN}I llama-cpp build info:noavx${RESET})
 	cp -rf backend/cpp/llama backend/cpp/llama-noavx
 	$(MAKE) -C backend/cpp/llama-noavx purge
-	CMAKE_ARGS+=-DLLAMA_AVX2=OFF
-	$(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
+	$(info ${GREEN}I llama-cpp build info:noavx${RESET})
+	CMAKE_ARGS="$(CMAKE_ARGS) -DLLAMA_AVX512=OFF -DLLAMA_AVX2=OFF" $(MAKE) VARIANT="llama-noavx" build-llama-cpp-grpc-server
 	cp -rfv backend/cpp/llama-noavx/grpc-server backend-assets/grpc/llama-cpp-noavx
 
 backend-assets/grpc/llama-cpp-fallback: backend-assets/grpc
-	$(info ${GREEN}I llama-cpp build info:fallback${RESET})
 	cp -rf backend/cpp/llama backend/cpp/llama-fallback
 	$(MAKE) -C backend/cpp/llama-fallback purge
-	CMAKE_ARGS+=-DLLAMA_F16C=OFF
-	CMAKE_ARGS+=-DLLAMA_AVX512=OFF
-	CMAKE_ARGS+=-DLLAMA_AVX2=OFF
-	CMAKE_ARGS+=-DLLAMA_FMA=OFF
-	$(MAKE) VARIANT="llama-fallback" build-llama-cpp-grpc-server
+	$(info ${GREEN}I llama-cpp build info:fallback${RESET})
+	CMAKE_ARGS="$(CMAKE_ARGS) -DLLAMA_F16C=OFF -DLLAMA_AVX512=OFF -DLLAMA_AVX2=OFF -DLLAMA_FMA=OFF" $(MAKE) VARIANT="llama-fallback" build-llama-cpp-grpc-server
 	cp -rfv backend/cpp/llama-fallback/grpc-server backend-assets/grpc/llama-cpp-fallback
 
 backend-assets/grpc/llama-ggml: sources/go-llama.cpp sources/go-llama.cpp/libbinding.a backend-assets/grpc

--- a/backend/cpp/llama/Makefile
+++ b/backend/cpp/llama/Makefile
@@ -43,29 +43,20 @@ llama.cpp:
 
 llama.cpp/examples/grpc-server: llama.cpp
 	mkdir -p llama.cpp/examples/grpc-server
-	cp -r $(abspath ./)/CMakeLists.txt llama.cpp/examples/grpc-server/
-	cp -r $(abspath ./)/grpc-server.cpp llama.cpp/examples/grpc-server/
-	cp -rfv $(abspath ./)/json.hpp llama.cpp/examples/grpc-server/
-	cp -rfv $(abspath ./)/utils.hpp llama.cpp/examples/grpc-server/
-	echo "add_subdirectory(grpc-server)" >> llama.cpp/examples/CMakeLists.txt
-## XXX: In some versions of CMake clip wasn't being built before llama.
-## This is an hack for now, but it should be fixed in the future.
-	cp -rfv llama.cpp/examples/llava/clip.h llama.cpp/examples/grpc-server/clip.h
-	cp -rfv llama.cpp/examples/llava/llava.cpp llama.cpp/examples/grpc-server/llava.cpp
-	echo '#include "llama.h"' > llama.cpp/examples/grpc-server/llava.h
-	cat llama.cpp/examples/llava/llava.h >> llama.cpp/examples/grpc-server/llava.h
-	cp -rfv llama.cpp/examples/llava/clip.cpp llama.cpp/examples/grpc-server/clip.cpp
+	bash prepare.sh
 
 rebuild:
-	cp -rfv $(abspath ./)/CMakeLists.txt llama.cpp/examples/grpc-server/
-	cp -rfv $(abspath ./)/grpc-server.cpp llama.cpp/examples/grpc-server/
-	cp -rfv $(abspath ./)/json.hpp llama.cpp/examples/grpc-server/
+	bash prepare.sh
 	rm -rf grpc-server
 	$(MAKE) grpc-server
 
-clean:
-	rm -rf llama.cpp
+purge:
+	rm -rf llama.cpp/build
+	rm -rf llama.cpp/examples/grpc-server
 	rm -rf grpc-server
+
+clean: purge
+	rm -rf llama.cpp
 
 grpc-server: llama.cpp llama.cpp/examples/grpc-server
 ifneq (,$(findstring sycl,$(BUILD_TYPE)))

--- a/backend/cpp/llama/Makefile
+++ b/backend/cpp/llama/Makefile
@@ -59,6 +59,7 @@ clean: purge
 	rm -rf llama.cpp
 
 grpc-server: llama.cpp llama.cpp/examples/grpc-server
+	@echo "Building grpc-server with $(BUILD_TYPE) build type and $(CMAKE_ARGS)"
 ifneq (,$(findstring sycl,$(BUILD_TYPE)))
 	bash -c "source $(ONEAPI_VARS); \
 	cd llama.cpp && mkdir -p build && cd build && cmake .. $(CMAKE_ARGS) && cmake --build . --config Release"	

--- a/backend/cpp/llama/prepare.sh
+++ b/backend/cpp/llama/prepare.sh
@@ -1,0 +1,20 @@
+#!/bin/bash
+
+cp -r CMakeLists.txt llama.cpp/examples/grpc-server/
+cp -r grpc-server.cpp llama.cpp/examples/grpc-server/
+cp -rfv json.hpp llama.cpp/examples/grpc-server/
+cp -rfv utils.hpp llama.cpp/examples/grpc-server/
+    
+if grep -q "grpc-server" llama.cpp/examples/CMakeLists.txt; then
+    echo "grpc-server already added"
+else
+    echo "add_subdirectory(grpc-server)" >> llama.cpp/examples/CMakeLists.txt
+fi
+
+## XXX: In some versions of CMake clip wasn't being built before llama.
+## This is an hack for now, but it should be fixed in the future.
+cp -rfv llama.cpp/examples/llava/clip.h llama.cpp/examples/grpc-server/clip.h
+cp -rfv llama.cpp/examples/llava/llava.cpp llama.cpp/examples/grpc-server/llava.cpp
+echo '#include "llama.h"' > llama.cpp/examples/grpc-server/llava.h
+cat llama.cpp/examples/llava/llava.h >> llama.cpp/examples/grpc-server/llava.h
+cp -rfv llama.cpp/examples/llava/clip.cpp llama.cpp/examples/grpc-server/clip.cpp

--- a/backend/go/llm/langchain/langchain.go
+++ b/backend/go/llm/langchain/langchain.go
@@ -4,6 +4,7 @@ package main
 // It is meant to be used by the main executable that is the server for the specific backend type (falcon, gpt3, etc)
 import (
 	"fmt"
+	"os"
 
 	"github.com/go-skynet/LocalAI/pkg/grpc/base"
 	pb "github.com/go-skynet/LocalAI/pkg/grpc/proto"
@@ -18,9 +19,14 @@ type LLM struct {
 }
 
 func (llm *LLM) Load(opts *pb.ModelOptions) error {
-	llm.langchain, _ = langchain.NewHuggingFace(opts.Model)
+	var err error
+	hfToken := os.Getenv("HUGGINGFACEHUB_API_TOKEN")
+	if hfToken == "" {
+		return fmt.Errorf("no huggingface token provided")
+	}
+	llm.langchain, err = langchain.NewHuggingFace(opts.Model, hfToken)
 	llm.model = opts.Model
-	return nil
+	return err
 }
 
 func (llm *LLM) Predict(opts *pb.PredictOptions) (string, error) {

--- a/core/http/app_test.go
+++ b/core/http/app_test.go
@@ -787,11 +787,11 @@ var _ = Describe("API test", func() {
 		})
 
 		It("returns errors", func() {
-			backends := len(model.AutoLoadBackends) + 1 // +1 for huggingface
 			_, err := client.CreateCompletion(context.TODO(), openai.CompletionRequest{Model: "foomodel", Prompt: testPrompt})
 			Expect(err).To(HaveOccurred())
-			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("error, status code: 500, message: could not load model - all backends returned error: %d errors occurred:", backends)))
+			Expect(err.Error()).To(ContainSubstring("error, status code: 500, message: could not load model - all backends returned error:"))
 		})
+
 		It("transcribes audio", func() {
 			if runtime.GOOS != "linux" {
 				Skip("test supported only on linux")

--- a/pkg/langchain/huggingface.go
+++ b/pkg/langchain/huggingface.go
@@ -2,6 +2,7 @@ package langchain
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/tmc/langchaingo/llms"
 	"github.com/tmc/langchaingo/llms/huggingface"
@@ -9,11 +10,16 @@ import (
 
 type HuggingFace struct {
 	modelPath string
+	token     string
 }
 
-func NewHuggingFace(repoId string) (*HuggingFace, error) {
+func NewHuggingFace(repoId, token string) (*HuggingFace, error) {
+	if token == "" {
+		return nil, fmt.Errorf("no huggingface token provided")
+	}
 	return &HuggingFace{
 		modelPath: repoId,
+		token:     token,
 	}, nil
 }
 
@@ -21,7 +27,7 @@ func (s *HuggingFace) PredictHuggingFace(text string, opts ...PredictOption) (*P
 	po := NewPredictOptions(opts...)
 
 	// Init client
-	llm, err := huggingface.New()
+	llm, err := huggingface.New(huggingface.WithToken(s.token))
 	if err != nil {
 		return nil, err
 	}

--- a/pkg/model/initializers.go
+++ b/pkg/model/initializers.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"slices"
 	"strings"
 	"time"
 
@@ -22,8 +23,11 @@ var Aliases map[string]string = map[string]string{
 }
 
 const (
-	LlamaGGML           = "llama-ggml"
-	LLamaCPP            = "llama-cpp"
+	LlamaGGML = "llama-ggml"
+	LLamaCPP  = "llama-cpp"
+
+	LLamaCPPFallback = "llama-cpp-fallback"
+
 	Gpt4AllLlamaBackend = "gpt4all-llama"
 	Gpt4AllMptBackend   = "gpt4all-mpt"
 	Gpt4AllJBackend     = "gpt4all-j"
@@ -62,6 +66,45 @@ ENTRY:
 			backends = append(backends, e.Name())
 		}
 	}
+
+	// order backends from the asset directory.
+	// as we scan for backends, we want to keep some order which backends are tried of.
+	// for example, llama.cpp should be tried first, and we want to keep the huggingface backend at the last.
+
+	// sets a priority list
+	// First has more priority
+	priorityList := []string{
+		// First llama.cpp and llama-ggml
+		LLamaCPP, LLamaCPPFallback, LlamaGGML, Gpt4All,
+	}
+	toTheEnd := []string{
+		// last has to be huggingface
+		LCHuggingFaceBackend,
+		// then bert embeddings
+		BertEmbeddingsBackend,
+	}
+	slices.Reverse(priorityList)
+	slices.Reverse(toTheEnd)
+
+	// order certain backends first
+	for _, b := range priorityList {
+		for i, be := range backends {
+			if be == b {
+				backends = append([]string{be}, append(backends[:i], backends[i+1:]...)...)
+				break
+			}
+		}
+	}
+	// make sure that some others are pushed at the end
+	for _, b := range toTheEnd {
+		for i, be := range backends {
+			if be == b {
+				backends = append(append(backends[:i], backends[i+1:]...), be)
+				break
+			}
+		}
+	}
+
 	return backends, nil
 }
 

--- a/pkg/model/initializers.go
+++ b/pkg/model/initializers.go
@@ -2,6 +2,7 @@ package model
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
@@ -9,7 +10,6 @@ import (
 	"time"
 
 	grpc "github.com/go-skynet/LocalAI/pkg/grpc"
-	"github.com/hashicorp/go-multierror"
 	"github.com/phayes/freeport"
 	"github.com/rs/zerolog/log"
 )
@@ -39,16 +39,22 @@ const (
 	LocalStoreBackend = "local-store"
 )
 
-var AutoLoadBackends []string = []string{
-	LLamaCPP,
-	LlamaGGML,
-	Gpt4All,
-	BertEmbeddingsBackend,
-	RwkvBackend,
-	WhisperBackend,
-	StableDiffusionBackend,
-	TinyDreamBackend,
-	PiperBackend,
+func backendPath(assetDir, backend string) string {
+	return filepath.Join(assetDir, "backend-assets", "grpc", backend)
+}
+
+func bakends(assetDir string) ([]string, error) {
+	entry, err := os.ReadDir(backendPath(assetDir, ""))
+	if err != nil {
+		return nil, err
+	}
+	var backends []string
+	for _, e := range entry {
+		if !e.IsDir() {
+			backends = append(backends, e.Name())
+		}
+	}
+	return backends, nil
 }
 
 // starts the grpcModelProcess for the backend, and returns a grpc client
@@ -99,7 +105,7 @@ func (ml *ModelLoader) grpcModel(backend string, o *Options) func(string, string
 				client = ModelAddress(uri)
 			}
 		} else {
-			grpcProcess := filepath.Join(o.assetDir, "backend-assets", "grpc", backend)
+			grpcProcess := backendPath(o.assetDir, backend)
 			// Check if the file exists
 			if _, err := os.Stat(grpcProcess); os.IsNotExist(err) {
 				return "", fmt.Errorf("grpc process not found: %s. some backends(stablediffusion, tts) require LocalAI compiled with GO_TAGS", grpcProcess)
@@ -243,7 +249,12 @@ func (ml *ModelLoader) GreedyLoader(opts ...Option) (grpc.Backend, error) {
 
 	// autoload also external backends
 	allBackendsToAutoLoad := []string{}
-	allBackendsToAutoLoad = append(allBackendsToAutoLoad, AutoLoadBackends...)
+	autoLoadBackends, err := bakends(o.assetDir)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug().Msgf("Loading from the following backends (in order): %+v", autoLoadBackends)
+	allBackendsToAutoLoad = append(allBackendsToAutoLoad, autoLoadBackends...)
 	for _, b := range o.externalBackends {
 		allBackendsToAutoLoad = append(allBackendsToAutoLoad, b)
 	}
@@ -271,10 +282,10 @@ func (ml *ModelLoader) GreedyLoader(opts ...Option) (grpc.Backend, error) {
 			log.Info().Msgf("[%s] Loads OK", b)
 			return model, nil
 		} else if modelerr != nil {
-			err = multierror.Append(err, modelerr)
+			err = errors.Join(err, modelerr)
 			log.Info().Msgf("[%s] Fails: %s", b, modelerr.Error())
 		} else if model == nil {
-			err = multierror.Append(err, fmt.Errorf("backend returned no usable model"))
+			err = errors.Join(err, fmt.Errorf("backend returned no usable model"))
 			log.Info().Msgf("[%s] Fails: %s", b, "backend returned no usable model")
 		}
 	}

--- a/pkg/model/initializers.go
+++ b/pkg/model/initializers.go
@@ -44,13 +44,20 @@ func backendPath(assetDir, backend string) string {
 	return filepath.Join(assetDir, "backend-assets", "grpc", backend)
 }
 
-func bakends(assetDir string) ([]string, error) {
+func backendsInAssetDir(assetDir string) ([]string, error) {
+	excludeBackends := []string{"local-store"}
 	entry, err := os.ReadDir(backendPath(assetDir, ""))
 	if err != nil {
 		return nil, err
 	}
 	var backends []string
+ENTRY:
 	for _, e := range entry {
+		for _, exclude := range excludeBackends {
+			if e.Name() == exclude {
+				continue ENTRY
+			}
+		}
 		if !e.IsDir() {
 			backends = append(backends, e.Name())
 		}
@@ -250,7 +257,7 @@ func (ml *ModelLoader) GreedyLoader(opts ...Option) (grpc.Backend, error) {
 
 	// autoload also external backends
 	allBackendsToAutoLoad := []string{}
-	autoLoadBackends, err := bakends(o.assetDir)
+	autoLoadBackends, err := backendsInAssetDir(o.assetDir)
 	if err != nil {
 		return nil, err
 	}

--- a/pkg/model/initializers.go
+++ b/pkg/model/initializers.go
@@ -15,9 +15,10 @@ import (
 )
 
 var Aliases map[string]string = map[string]string{
-	"go-llama":       LLamaCPP,
-	"llama":          LLamaCPP,
-	"embedded-store": LocalStoreBackend,
+	"go-llama":              LLamaCPP,
+	"llama":                 LLamaCPP,
+	"embedded-store":        LocalStoreBackend,
+	"langchain-huggingface": LCHuggingFaceBackend,
 }
 
 const (
@@ -34,7 +35,7 @@ const (
 	StableDiffusionBackend = "stablediffusion"
 	TinyDreamBackend       = "tinydream"
 	PiperBackend           = "piper"
-	LCHuggingFaceBackend   = "langchain-huggingface"
+	LCHuggingFaceBackend   = "huggingface"
 
 	LocalStoreBackend = "local-store"
 )

